module/Finna/tests/unit-tests/src/FinnaTest/Controller/CoverControllerTest.php

```diff
@@ -0,0 +1,403 @@
+<?php
+
+/**
+ * CoverController test class
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace FinnaTest\Controller;
+
+use Finna\Controller\CoverController;
+use Finna\Cover\Loader;
+use Finna\Db\Service\AccessTokenService;
+use Finna\File\Loader as FileLoader;
+use Finna\RecordDriver\SolrLido;
+use Generator;
+use Laminas\Db\ResultSet\ResultSet;
+use Laminas\Http\Headers;
+use Laminas\Http\PhpEnvironment\Response;
+use Laminas\Stdlib\Parameters;
+use VuFind\Config\Config;
+use VuFind\Config\PluginManager;
+use VuFind\Cover\CachingProxy;
+use VuFind\Cover\Loader as CoverLoader;
+use VuFind\Db\Row\AccessToken as RowAccessToken;
+use VuFind\Db\Table\AccessToken;
+use VuFind\Http\PhpEnvironment\Request;
+use VuFind\Record\Loader as RecordLoader;
+use VuFind\RecordDriver\Missing;
+use VuFind\Session\Settings;
+use VuFindTest\Feature\FixtureTrait;
+
+/**
+ * CoverController test class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Juha Luoma <juha.luoma@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class CoverControllerTest extends \PHPUnit\Framework\TestCase
+{
+    use FixtureTrait;
+
+    /**
+     * Mock container
+     *
+     * @var \VuFindTest\Container\MockContainer
+     */
+    protected $container;
```

Comment:
Mä luulen, että tätä voisi yksinkertaistaa vielä enemmän. Koska containeria käytetään käytännössä vain constructorin parametreissa, voisi ne korvata suoraan mockeilla container->get():ien sijaan. Eli esim. `$this->container->get(CoverLoader::class)` => `$this->container->createMock(Loader::class)`.

Tarvittaessa yliajettavat luokat sitten parametreina getCoverController-metodille eikä niin, että laitetaan containeriin jotain uutta. Tällä tavalla ei tarvita jaettua containeria ja saadaan varmistettua, että testit ovat itsenäisiä, eikä yksittäinen testi muuta containerin sisältöä.